### ISSUE TITLE
Problem with GQA question ID

### ISSUE THREAD

**Issue by pkhdipraja** [facebookresearch/vilbert-multi-task#59]
https://github.com/facebookresearch/vilbert-multi-task/blob/6c1d98c3b808910c64d88c39dfb9e46c6b6abc99/vilbert/datasets/gqa_dataset.py#L28

Converting string to int for GQA datasets can cause possible mismatch for the prediction as there are question IDs such as `01827492` and `1827492` in balanced validation sets, which is problematic for evaluation using the official script.

**Comment by vedanuj** [comment]
For validation you should use the `https://github.com/facebookresearch/vilbert-multi-task/blob/master/eval_tasks.py` script.

**Comment by pkhdipraja** [comment]
The script works when used to produce prediction for EvalAI test split, but when it is used to output prediction for val split and evaluate it with the official script `https://cs.stanford.edu/people/dorarad/gqa/evaluate.html`, it didn't work due to question ID mismatch.

**Comment by aurooj** [comment]
I am also facing the question id mismatch. @pkhdipraja, were you able to fix this problem?

**Comment by pkhdipraja** [comment]
@aurooj although my code is messy, I do have a quick hack if you want to evaluate using the official script: `https://github.com/pkhdipraja/joint-multimodal-embeddings`. However, I only tested it on VQA2.0 and GQA.

**Comment by aurooj** [comment]
That's great. Thanks a lot!

**Comment by aurooj** [comment]
@pkhdipraja I have a question regarding your code for extracting output attention from vilbert model. You use the following line:
`last_layer_att_score = torch.squeeze(att_score[1][-1]['attn'][:, 7, 0, :])  # batch_size, att_head, target_num_feat, source_num_feat -> use all att head and CLS as target`

I am not sure if the comment matches the dimensions of att_score. More specifically, what is the dimension 7 here? My understanding is that it's the 8th head. Can you please confirm?
Thanks!

**Comment by pkhdipraja** [comment]
Yes, I'm using that to get the attention score for the 8th head, with CLS token as the query.

**Comment by aurooj** [comment]
Okay, thanks!
